src/core/qgsfield.cpp

```diff
@@ -589,12 +589,61 @@ bool QgsField::convertCompatible( QVariant &v, QString *errorMessage ) const
     return false;
   }
 
+  if ( d->typeName == "geometry" && v.type() == QVariant::String )
+  {
+    // TODO: expose this as a "canonical text input" method of
+    // QgsReferencedGeometry class
+    auto parts = v.toString().split( " [" );
+    auto wkt = parts[0];
+    // TODO: handle more string to mean NULL ?
```

Comment:
I don't understand this first test. Why don't try to use directly `QgsGeometry::fromWkt`?

Reply:
`Point (1 1) [EPSG:4326 - WGS 84]` - not similar to WKT

Reply:
Oh, ok!
It is a (yet another) normalized way to write this EWKT-like geometry ?

Reply:
this format was introduced in #45127

Reply:
... which was meant to be representation only ...